adoc/chapters/programming_interface.adoc

```diff
@@ -4918,8 +4918,10 @@ STL-based libraries (e.g, Intel's TBB provides an allocator).
 ==== Default allocators
 
 A default allocator is always defined by the implementation.
-For allocations greater than size zero, when successful it is guaranteed to
-return non-[code]#nullptr# and new memory positions every call.
+For successful allocations of size greater than zero, the default allocator must
+return a pointer to a contiguous, exclusive region of memory of at least the
+requested size.
+For unsuccessful allocations, the default allocator must return [code]#nullptr#.
```

Comment:
> For unsuccessful allocations, the default allocator must return [code]#nullptr#.

I think this is not the way allocators normally handle out-of-memory.  A C++ allocator normally throws an exception in this case.

Reply:
I agree that [named requirement: Allocator](https://en.cppreference.com/w/cpp/named_req/Allocator) implies that the only alternative an allocator has to providing the requested memory is throwing.

A few lines earlier in the SYCL spec (L 4892), it says:

```quote
When an allocator returns a `nullptr`, the runtime cannot allocate data on the host.
```

So I guess either the SYCL spec expect more C-like behavior (in which case default allocators are not actually `named requirement Allocator`s), or default allocators throw on failure, in which case an allocator will actually throw when the runtime can't allocate data on the host. So one sentence or the other needs to change in the SYCL spec.